### PR TITLE
Fix for VB-3651 - add prisonCode as a parameter when getting visits by session template reference.

### PR DESCRIPTION
**Pull request by dhirajshettymoj**
## What does this pull request do?

adds prisonCode as a parameter when getting visits by session template reference.
## What is the intent behind these changes?

VB-3651 bug fix

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/controller/VisitController.kt [hunk 1]
```diff
@@ -360,6 +360,10 @@ class VisitController(
     )
     @NotNull
     visitStatusList: List<VisitStatus>,
+    @Schema(name = "prisonCode", description = "Filter results by prison id/code", example = "MDI", required = true)
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/controller/VisitController.kt:363]: add prisonCode parameter

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt [hunk 2]
```diff
@@ -186,7 +186,8 @@ interface VisitRepository : JpaRepository<Visit, Long>, JpaSpecificationExecutor
       "(:#{#visitStatusList} is null OR v.visitStatus in :visitStatusList) AND " +
       "(:#{#visitRestrictions} is null OR v.visitRestriction in :visitRestrictions) AND " +
       "v.sessionSlot.slotDate >= :fromDate  AND " +
-      "v.sessionSlot.slotDate <= :toDate " +
+      "v.sessionSlot.slotDate <= :toDate AND " +
+      "v.prison.code = :prisonCode " +
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt:190]: add prison code parameter to SQL query to ensure visits are only being returned for that prison

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt [hunk 3]
```diff
@@ -218,14 +220,16 @@ interface VisitRepository : JpaRepository<Visit, Long>, JpaSpecificationExecutor
...
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt:224]: add prison code parameter to SQL query to ensure visits are only being returned for that prison

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/visit/VisitsBySessionTemplateFilterTest.kt [hunk 4]
```diff
@@ -176,13 +189,14 @@ class VisitsBySessionTemplateFilterTest : IntegrationTestBase() {
...
-  fun `when session template reference passed is null only session template reference null records are returned`() {
+  fun `when session template reference passed is null only session template reference null records are returned for that prison`() {
     // Given
     val fromDate = LocalDate.now()
     val toDate = startDate.plusDays(2)
+    val prisonCode = "MDI"
 
     // When
-    val params = getVisitsBySessionTemplateQueryParams(sessionTemplateReference = null, fromDate = fromDate, toDate = toDate, visitStatus = listOf(BOOKED, CANCELLED)).joinToString("&")
+    val params = getVisitsBySessionTemplateQueryParams(sessionTemplateReference = null, fromDate = fromDate, toDate = toDate, visitStatus = listOf(BOOKED, CANCELLED), prisonCode = prisonCode).joinToString("&")
```
Comment (dhirajshettymoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/visit/VisitsBySessionTemplateFilterTest.kt:199]: updated tests to ensure visits are being returned only for that prison

**Review by tpmcgowan — APPROVED** [review]
LGTM

**Pull request merged**